src/app/bookshelf/AddToBookshelf.tsx

```diff
@@ -0,0 +1,123 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import styles from "./AddToBookshelf.module.css";
+import { useForm } from "react-hook-form";
+import { AddBookshelfBook, BookStatus } from "@/interfaces/BookshelfBook";
+import { fetchBookByIsbn } from "@/api/fetchBookByIsbn";
+import { useAddBookToBookshelfMutation } from "@/api/bookshelfApi";
+
+import Input from "@/components/inputs/Input";
+import Select from "@/components/inputs/Select";
+
+interface FetchStatus {
+  success: boolean;
+  message: string;
+}
+
+export default function AddToBookshelf() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<AddBookshelfBook>();
+  const [formError, setFormError] = useState("");
+  const [fetchStatus, setFetchStatus] = useState<FetchStatus | undefined>(
+    undefined
+  );
+
+  const [addBookToBookshelf, { isLoading, isSuccess, isError }] =
+    useAddBookToBookshelfMutation();
+
+  useEffect(() => {
```

Comment:
This `useEffect` will add to the re-render cycle of the component (it'll get re-calculated and then trigger another re-render). 

As the result of fetching can be assumed directly from the value of both variables used in it, I'd suggest just creating a helper function to return this payload/transform `isSuccess` and `isError` into this payload instead of creating `fetchStatus` and name it e.g. `getFetchStatus`.